tests/core/mempool/test_mempool_manager.py

```diff
@@ -207,6 +225,46 @@ def test_seconds_absolute(
             == expected_error
         )
 
+    @pytest.mark.parametrize(
+        "value,expected_error",
+        [
+            # the coin's birth height is
+            (9, Err.ASSERT_MY_BIRTH_HEIGHT_FAILED),
+            (10, None),
+            (11, Err.ASSERT_MY_BIRTH_HEIGHT_FAILED),
+        ],
+    )
+    def test_assert_my_birth_height(
+        self,
+        value: uint32,
+        expected_error: Optional[Err],
+    ) -> None:
+        conds = make_test_conds(birth_height=value)
+        assert (
+            mempool_check_time_locks(self.REMOVALS, conds, self.PREV_BLOCK_HEIGHT, self.PREV_BLOCK_TIMESTAMP)
+            == expected_error
+        )
+
+    @pytest.mark.parametrize(
+        "value,expected_error",
+        [
+            # the coin's birth timestamp is
+            (9999, Err.ASSERT_MY_BIRTH_SECONDS_FAILED),
+            (10000, None),
+            (10001, Err.ASSERT_MY_BIRTH_SECONDS_FAILED),
+        ],
+    )
+    def test_assert_my_birth_seconds(
+        self,
+        value: uint64,
+        expected_error: Optional[Err],
+    ) -> None:
+        conds = make_test_conds(birth_seconds=value)
+        assert (
+            mempool_check_time_locks(self.REMOVALS, conds, self.PREV_BLOCK_HEIGHT, self.PREV_BLOCK_TIMESTAMP)
```

Comment:
is it possible to provide actual coin birth height when returning the error to simplify debugging?

Reply:
you mean it could return something like a `ValidationErr()` with the error code and include additional fields for the value? the function checks the `ASSERT_HEIGHT_*` and `ASSERT_SECONDS_*` conditions as well (and, as you can imagine, will check `ASSERT_BEFORE_*` conditions in a future patch).

This function is used in two places: the mempool, to gate transactions that aren't valid and in the block body validation for consensus, validating blocks. Any value returned from this function would need to be returned out of those contexts too, before reaching the user that's debugging. It's not obvious how to do that. logging might work, but on the other hand, when running a node you don't necessarily want to log just because one of your peers is submitting invalid transactions.